### PR TITLE
Avoid loading files' local variables [Bug #17623]

### PR DESCRIPTION
**Pull request by marcandre**
https://bugs.ruby-lang.org/issues/17623

### CODE REVIEW

**Comment by marcandre** [comment]
This also fixes the same issue with `irb --single` that must have existed for years

**Review by jeremyevans — APPROVED** [review]
I think this is a good way to fix the issue.

**Comment by eregon** [comment]
This seems fine to me too given it assigns to `IRB::TOPLEVEL_BINDING` and does not override `::TOPLEVEL_BINDING`.

File: exe/irb [hunk 1]
```diff
@@ -8,4 +8,5 @@
 
 require "irb"
 
+IRB::TOPLEVEL_BINDING = binding
```
Comment (aycabta) [inline exe/irb:11]: I believe that this is a good solution to fix the problem but this assignment is so far from lib/irb/workspace.rb. Would you move the assignment to the top-level of lib/irb/workspace.rb? And, `TOPLEVEL_BINDING` is confusing with the existing constant, so I think a different name would be better.

**Review by aycabta — CHANGES_REQUESTED** [review]

File: exe/irb (hunk 1 above)
Comment (eregon) [reply exe/irb:11]: Agreed, a different constant name would be much less confusing.

**Pull request merged**